### DISCUSSION TITLE
webhook integration with slack

### DISCUSSION THREAD

**Discussion by jinaloo7** [netbox-community/netbox#7982] [Q&A] [answered]
Hello,

I am trying to use netbox webhook integration with slack and it shows no result.where I can find that netbox job is initiated and logs related to failure success?

Netbox version 2.10.6

**Comment by candlerb** [comment, marked as answer]
Logs are in the admin section of the Netbox web interface.  I don't have the older 2.10 around to test, but in 3.0 there are sections called "Job results" and "Background tasks".  I think webhooks are "Background tasks".  It will show a number of successful or failed tasks; click on the number and it will list them individually.

**Reply by jinaloo7** [reply]
Ahh I can see them now and it says there that

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/rq/worker.py", line 1043, in perform_job
    rv = job.perform()
  File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 814, in perform
    self._result = self._execute()
  File "/usr/local/lib/python3.7/site-packages/rq/job.py", line 837, in _execute
    result = self.func(*self.args, **self.kwargs)
  File "/opt/netbox/netbox/extras/webhooks_worker.py", line 82, in process_webhook
    response.status_code, response.content
requests.exceptions.RequestException: Status 400 returned with content 'b'invalid_payload'', webhook FAILED to process.

looks like payload sent to slack is rejected.How can I set payload from UI or from code itself for sending slack request any idea?

**Reply by candlerb** [reply]
You can control the body using a [jinja2 template](https://netbox.readthedocs.io/en/stable/additional-features/webhooks/#jinja2-template-support).

**Reply by jinaloo7** [reply]
Thanks I missed that but one thing is that it is now not failing but staying status as queued only

do you know what might cause that and how to fix that?

<img width="1074" alt="Screen Shot 2021-12-07 at 3 22 06 PM" src="https://user-images.githubusercontent.com/26171542/145006735-adbe58ed-21d3-4db6-a776-eaa7c2039394.png">
.

**Reply by candlerb** [reply]
Maybe your template is invalid?  Do you see anything in the process error output?

```
journalctl -eu netbox-rq
```

**Reply by jinaloo7** [reply]
No it is working with template locally but not with deployed one.Is there a parameter I need to set to dequeue jobs? Seems like something I have closed or forgot to turn on.

**Reply by candlerb** [reply]
> Is there a parameter I need to set to dequeue jobs? 

No.  As long as netbox-rq is running, it will dequeue jobs.

I do not understand what you mean by "it is working with template locally but not with deployed one"

**Reply by jinaloo7** [reply]
I mean I have one environment which runs in my cloud and one in my local docker pods.So the one deployed in cloud does not work and doesn't dequeuing jobs.

**Reply by candlerb** [reply]
Then probably netbox-rq isn't running there, or is giving errors when it tries to dequeue jobs.  I'm afraid you'll have to debug this yourself.  It's the same version of Netbox (presumably) and therefore it's something about your environment that's different.

**Reply by jinaloo7** [reply]
Correct.I need to see what is going on there.Thanks for help though.
